### ISSUE TITLE
Apply Notification "users" and "user_groups" as variable

### ISSUE THREAD

**Issue by n00by** [voxpupuli/puppet-icinga2#208]
**What I want to do**

from icinga2 docs, 4.2.3.4. notifications.conf - [https://docs.icinga.com/icinga2/latest/doc/module/icinga2/toc#!/icinga2/latest/doc/module/icinga2/chapter/configuring-icinga2-first-steps#notifications-conf](url)

```
apply Notification "mail-icingaadmin" to Host {
  import "mail-host-notification"

  user_groups = host.vars.notification.mail.groups
  users = host.vars.notification.mail.users

  assign where host.vars.notification.mail
}
```

**That's my config**

```
icinga2::object::notification { 'mail-host-notification_apply':
  notification_name => 'mail-icingaadmin',
  import            => [ 'mail-host-notification' ],
  apply             => true,
  user_groups       => 'host.vars.notification.mail.groups',
  users             => 'host.vars.notification.mail.users',
  assign            => [ 'host.vars.notification.mail' ],
  target            => '/etc/icinga2/conf.d/notifications.conf',
}
```

**What I get**

 ```
apply Notification "mail-icingaadmin" to Host {
  import "mail-host-notification"

  users = [ host.vars.notification.mail.users, ]
  user_groups = [ host.vars.notification.mail.groups, ]
  assign where host.vars.notification.mail
}
```

Problem is that icinga doesn't identify variables as variables when set in square brackets.

**Possible solution**

Check if the parameter is a variable in the puppet code. So `any2array` "users" and "user_groups" solely when not a variable.

**Comment by n00by** [comment]
Hi, this issue is not fixed in v1.0.2.

Values for "users" and "user_groups" must be a string if you want to use them as variables. Therefore both string and array should be possible:

```
if !is_array($users) { validate_string($users) }
if !is_string($users) { validate_array($users) }
```

To keep things tidy there should be a check if a user really wants to use a variable if a string is used. So an additional check is necessary, sth. like `if is_string($users) and !is_variable($users) then any2array($users)`.

Or just add to the documentation to use a string for setting a variable, else an array.

**Comment by lbetz** [comment]
Done. Did the same for states and types attribute.
